src/main/java/seedu/address/model/person/Person.java

```diff
@@ -64,10 +64,34 @@ public Set<PublicAddress> getPublicAddressesByNetwork(Network network) {
         return publicAddresses.getOrDefault(network, new HashSet<>());
     }
 
+    public void setPublicAddressesByNetwork(Network network, HashSet<PublicAddress> addresses) {
+        if (publicAddresses.containsKey(network)) {
+            this.publicAddresses.put(network, addresses);
+        }
+    }
+
+
+    /**
+     * Gets the current Public address map
+     * Should be replaced with a better method in the future
+     *
+     * @return publicAddresses
+     */
     public Map<Network, Set<PublicAddress>> getPublicAddresses() {
-        return Collections.unmodifiableMap(publicAddresses);
+        return this.publicAddresses;
```

Comment:
Same thing here, much safer to return copy / unmodifiable.

Reply:
Fixed in commit 39c5ce2c8be90a5c6695fc5696ac1f59b783b4e8